drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -975,25 +975,65 @@ def _get_user_state(self):
     def _get_correct_state(self):
         """
         Returns one of the possible correct states for the configured data.
+
+        Items dropped by user in the correct zones are retained in those zones.
+        If there are more than one correct zone of a group of items which are dropped in
+        wrong zones, they are equally distributed among the correct zones.
         """
         state = {}
         items = copy.deepcopy(self.data.get('items', []))
+
+        zone_count = {}
+        correct_items = []
+
+        def _get_display_zone(zone_count, zones):
+            """
+            Returns the zone with the least number of items.
+            """
+            display_zone = None
+            for zone in zones:
+                # Check if no item is present in zone
+                if zone not in zone_count:
+                    zone_count[zone] = 1
+                    return zone
+                # Check if the current zone has the least number of items till now
+                elif not display_zone or zone_count[display_zone] > zone_count[zone]:
+                    display_zone = zone
+
+            zone_count[display_zone] += 1
+            return display_zone
+
+        # Set states of all items dropped in correct zones
+        for item_id in self.item_state:
+            if self.item_state[item_id]['correct']:
+                state[item_id] = self.item_state[item_id]
+                correct_items.append(item_id)
+
+                zone = self.item_state[item_id]['zone']
+                if zone not in zone_count:
+                    zone_count[zone] = 1
+                else:
+                    zone_count[zone] += 1
```

Comment:
Nit: we can use a [Counter](https://docs.python.org/3.8/library/collections.html#collections.Counter) for this purpose. It supports `zone_count[zone] += 1` even if an item does not exist there yet.

It could also simplify `if zone not in zone_count` to `if not zone_count[zone]`, but this is even less important.

Reply:
Thanks for pointing this out. Using `Counter` makes the logic simpler.